aquadoggo/src/replication/manager.rs

```diff
@@ -176,47 +200,131 @@ where
             SessionState::Pending => {
                 if &self.local_peer < remote_peer {
                     // Drop our pending session
-                    self.remove_session(remote_peer, &session.id);
+                    debug!(
+                        "Drop pending outbound session and process inbound session request with duplicate id {}",
+                        existing_session.id
+                    );
+                    self.remove_session(remote_peer, &existing_session.id);
 
                     // Accept the inbound request
-                    true
+                    Ok(true)
                 } else {
                     // Keep our pending session, ignore inbound request
-                    false
+                    debug!(
+                        "Ignore inbound request and keep pending outbound session with duplicate id {}",
+                        existing_session.id
+                    );
+                    Ok(false)
                 }
             }
-            _ => return Err(ReplicationError::DuplicateInboundRequest(session.id)),
-        };
+            SessionState::Established => Err(
+                DuplicateSessionRequestError::InboundEstablishedSession(existing_session.id),
+            ),
+            SessionState::Done => Err(DuplicateSessionRequestError::InboundDoneSession(
+                existing_session.id,
+            )),
+        }?;
 
         let mut all_messages: Vec<SyncMessage> = vec![];
 
         if accept_inbound_request {
+            debug!(
+                "Accept duplicate session request with id {} for peer {:?}",
+                existing_session.id, remote_peer
+            );
             let messages = self
                 .insert_and_initialize_session(
                     remote_peer,
-                    &session.id,
+                    &existing_session.id,
                     target_set,
-                    &session.mode(),
+                    &existing_session.mode(),
                     false,
                 )
                 .await;
-            all_messages.extend(to_sync_messages(session.id, messages));
-
-            // If we dropped our own outbound session request regarding a different target set, we
-            // need to re-establish it with another session id, otherwise it would get lost
-            if session.target_set() != *target_set {
-                let messages = self
-                    .initiate_session(remote_peer, target_set, &session.mode())
-                    .await?;
-                all_messages.extend(messages)
+            all_messages.extend(to_sync_messages(existing_session.id, messages));
+
+            // @TODO: Do we want to re-initiate the dropped session if it was concerning a
+            // different target set?
+        }
+
+        Ok(SyncResult {
+            messages: all_messages,
+            is_done: false,
+        })
+    }
+
+    async fn handle_duplicate_target_set(
```

Comment:
Added this method for handling session requests with duplicate `TargetSet`s in the same way as we do for those with duplicate session ids.

Reply:
Sure these two methods could be refactored to avoid some code duplication too....